tests/rptest/tests/admin_uuid_operations_test.py

```diff
@@ -408,3 +408,92 @@ def test_uuid_override_multinode(self, mode, force):
+        """
+        Test that we can force a UUID override in situations where the cluster
+        is healthy and the target node already has a node ID assigned
+        """
+        # create a topic so that the cluster is not completely empty
+        RpkTool(self.redpanda).create_topic("foo", 10, 3)
+
+        to_stop = self.redpanda.nodes[0]
+        initial_to_stop_id = self.redpanda.node_id(to_stop)
+
+        self._restart_node(to_stop, drop_disk=True)
+
+        # wait for the node to join with new ID
+        uuids = wait_until_result(
+            lambda: self._uuids_updated(),
+            timeout_sec=30,
+            backoff_sec=2,
+            err_msg="Node was unable to join the cluster")
+
+        old_uuid = None
+        for n in uuids:
+            id = n['node_id']
+            if id == initial_to_stop_id:
+                old_uuid = n['uuid']
+
+        assert old_uuid is not None, "Old uuid unexpectedly None"
+
+        current_uuid = self.admin.get_broker_uuid(to_stop)['node_uuid']
+
+        THE_OVERRIDE = f"{current_uuid} -> ID: '{initial_to_stop_id}' ; UUID: '{old_uuid}'"
+        if mode == TestMode.CFG_OVERRIDE:
+            self.logger.debug(
+                f"Override with known-good uuid/id via node config: {THE_OVERRIDE}"
+            )
+            self._restart_node(
+                to_stop,
+                dict(node_id_overrides=[
+                    dict(current_uuid=current_uuid,
+                         new_uuid=old_uuid,
+                         new_id=initial_to_stop_id,
+                         force=force)
+                ], ),
+                drop_disk=False,
+            )
+        elif mode == TestMode.CLI_OVERRIDE:
+            self.logger.debug(
+                f"Override with known-good uuid/id via command line options: {THE_OVERRIDE}"
+            )
+            self._restart_node(
+                to_stop,
+                extra_cli=[
+                    "--node-id-overrides",
+                    f"{current_uuid}:{old_uuid}:{initial_to_stop_id}{'/force' if force else ''}",
+                ],
+                drop_disk=False,
+            )
+
+        def wait_for_override():
+            wait_until(lambda: self.admin.get_broker_uuid(to_stop)['node_id']
+                       == initial_to_stop_id,
+                       timeout_sec=30,
+                       backoff_sec=2,
+                       err_msg=f"{to_stop.name} did not take the ID override")
+
+            wait_until(
+                lambda: self.admin.get_broker_uuid(to_stop)['node_uuid'
+                                                            ] == old_uuid,
+                timeout_sec=30,
+                backoff_sec=2,
+                err_msg=f"{to_stop.name} did not take the UUID override")
+
+        if force:
+            wait_for_override()
+        else:
+            with expect_exception(TimeoutError, lambda _: True):
+                wait_for_override()
```

Comment:
for completeness, I'm wondering if we can restart the broker with the same config and no changes take effect (because of UUID mismatch).

Reply:
done